File: promotion/user-service/src/main/java/com/study/userservice/repository/UserLoginHistoryRepository.java
package com.study.userservice.repository;

import com.study.userservice.entity.User;
import com.study.userservice.entity.UserLoginHistory;
import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserLoginHistoryRepository extends JpaRepository<UserLoginHistory, Long> {
  List<UserLoginHistory> findByUserOrderByLoginTimeDesc(User user);
}

File: promotion-kotlin/user-service-kotlin/src/main/kotlin/com/study/userservicekotlin/repository/UserLoginHistoryRepository.kt
package com.study.userservicekotlin.repository

import com.study.userservicekotlin.entity.User
import com.study.userservicekotlin.entity.UserLoginHistory
import org.springframework.data.jpa.repository.JpaRepository

interface UserLoginHistoryRepository: JpaRepository<UserLoginHistory, Long> {
    fun findByUserOrderByLoginTimeDesc(user: User): List<UserLoginHistory>
}
